src/store/channels-list/saga.ts

```diff
@@ -409,12 +413,17 @@ export function* otherUserJoinedChannel(roomId: string, user: User) {
     return;
   }
 
-  // TODO: Fetch user from zOS if we don't know about them yet
+  if (user.userId === user.matrixId) {
+    user = yield select(userByMatrixId, user.matrixId);
+  }
+
   if (!channel.otherMembers.includes(user.userId)) {
+    const otherMembers = [...channel.otherMembers, user];
     yield put(
       receiveChannel({
         id: channel.id,
-        otherMembers: [...channel.otherMembers, user],
+        isOneOnOne: otherMembers.length === 1,
```

Comment:
This is slightly incorrect. If it's a `Channel` and not a `Conversation` then it's not considered a One on One.

Reply:
Upon second thought, I do think at this level this is actually just wrong. This would consider old group conversations where everyone but 2 people left as one on ones and that's not right. Those are still group conversations. This field should be provided by the matrix client where it knows the details of these types of things. (you can see where isOneOnOne is calculated when loading rooms)

Reply:
> This is slightly incorrect. If it's a Channel and not a Conversation then it's not considered a One on One.

will update this to only set `isOneOnOne` in case if `isChannel` is `false`.

Reply:
> This field should be provided by the matrix client where it knows the details of these types of things. (you can see where isOneOnOne is calculated when loading conversations)

This is where i am confused. We don't "load" the conversations while (or after) creating the conversation. And while creating the conversation immediately we just have the raw conversation with 0 members. Then we get the events that a member has joined channel, and an "other" member has joined channel. 

The only place we calculate the conversation's `isOneOnOne` is while mapping them (which happens while loading all conversations). Which we don't do here. I guess we could "force" the conversation list to reload here as well to get proper values. But i guess since it's all event based, i don't know when the events will "stop" (after the 1st user joins, 2nd..or 5th? ..etc). If we reload the conversation list sooner than the events haven't stopped coming through, then we'll still end up with a weird state :(

Reply:
re: group conversation, i think we shouldn't have an issue with members leaving the channel (as in that case only the LEAVE event will be triggered, not this one). But i don't know about the case if i start a group conversation with 3 people, in that case this might be wrong. I will test that also

Reply:
create group conversation:

https://github.com/zer0-os/zOS/assets/33264364/6acb6381-43b1-4e91-9679-1c4da3ea3dce

Reply:
> This is slightly incorrect. If it's a Channel and not a Conversation then it's not considered a One on One.

> Even there, I think that's wrong. A "Channel" should never be considered one on one.

@dalefukami found something interesting along this note. Currently we're **always** setting `isChannel == true` while creating a conversation. Currently [this](https://github.com/zer0-os/zOS/blob/d61983e31b96eb240bada93dddac6f29d63138d9/src/lib/chat/matrix-client.ts#L413) event is fired when after we create the room, but this maps the channel generally and always sets `isChannel` to be true. So while creating the conversation, at all event levels, we're assuming this is a channel (& not a conversation) 🤔